src/components/Layout/RightPaneLayout/RightPaneLayout.tsx

```diff
@@ -0,0 +1,52 @@
+import { AnimatePresence, motion } from "framer-motion";
+import React, { ReactElement } from "react";
+import { Css, Palette, px } from "../../../Css";
+import { useRightPaneContext } from "./RightPaneContext";
+
+export function RightPaneLayout({ children, paneBgColor }: { children: ReactElement; paneBgColor?: Palette }) {
+  const { isRightPaneOpen, rightPaneContent } = useRightPaneContext();
+
+  // Todo pass as prop?
+  const paneWidth = 450;
+
+  return (
+    <div css={Css.h100.df.$}>
+      <>
+        <motion.div
+          layout="position"
+          key="rightPaneLayoutPageContent"
+          css={Css.h100.if(!!isRightPaneOpen).overflowX("scroll").mr3.if(!isRightPaneOpen).overflowX("unset").$}
+          initial={"closed"}
+          animate={!isRightPaneOpen ? "closed" : "open"}
+          variants={{
+            open: { width: `calc(100% - ${paneWidth + 24}px)` },
+            closed: { width: "100%" },
+          }}
+          transition={{ ease: "linear", duration: 0.2 }}
+        >
+          {children}
+        </motion.div>
+        <AnimatePresence>
+          {isRightPaneOpen && (
+            <motion.div
+              layout="position"
+              key="rightPane"
+              data-testid="rightPaneContent"
+              css={Css.bgColor(paneBgColor).h100.maxw(px(paneWidth)).w100.df.fdc.relative.$}
+              // Keeping initial x to offset pane width and space between panel and page content
+              initial={{ x: paneWidth + 24 }}
+              animate={{ x: 0 }}
+              // Custom transitions settings for the translateX animation
+              transition={{ ease: "linear", duration: 0.2 }}
+              exit={{ transition: { ease: "linear", duration: 0.2 }, x: paneWidth }}
```

Comment:
**thought:** If I'm reading this correctly, it seems like the Right Pane is immediately rendered once `isRightPaneOpen` is true, but it's rendered off screen. Then uses a `translateX` transition to bring it into the viewport, is that right?  If so, I'm thinking that would cause an unnecessary horizontal scrollbar until the transition completes and the pane is completely in the viewport.

Reply:
Maybe `AnimatePresence` solves that? I dunno.

Reply:
Doesn't appear to solve it:
https://www.loom.com/share/6d136b819ea645e7853226ae8cc40ea7

Reply:
was able to resolve this by using `overflowXHidden`